docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/migrate-front-end.md

```diff
@@ -151,6 +155,201 @@ export default {
+
+::: details Example of adding reducers
+
+```js
+// path: ./src/plugins/my-plugin/admin/src/index.js
+
+import myReducer from './components/MyCompo/reducer';
+import myReducer1 from './components/MyCompo1/reducer';
+import pluginId from './pluginId';
+
+const reducers = {
+  [`${pluginId}_reducer`]: myReducer,
+  [`${pluginId}_reducer1`]: myReducer1,
+};
+
+export default {
+  register(app) {
+    app.addReducers(reducers);
+
+    app.registerPlugin({
+      id: pluginId,
+      initializer: Initializer,
+      isReady: false,
+      name,
+    });
+  },
+ }
+}
+```
+
+:::
+
+## Adding injection zones
+
+::: strapi v3/v4 comparison
+A Strapi v3 plugin can inject components into the Content Manager's Edit view, using the `registerField()` method or the `useStrapi` hook within the `Initializer` component.
+
+In Strapi v4, a plugin can inject components into several locations of the Content Manager using the [Injection Zones API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#injection-zones-api).
+:::
+
+To migrate to Strapi v4, make sure components are injected using Strapi v4 [Injection Zones API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#injection-zones-api). Depending on where the component should be injected, use:
+
+- either the `injectContentManagerComponent()` method to inject into [predefined injection zones](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#using-predefined-injection-zones) of the Content Manager
+- or the `injectComponent()` method to inject into [custom zones](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.md#creating-a-custom-injection-zone)
+
+::: details Example of injecting a component into the Content Manager's Edit view
+
+```jsx
+// path:  ./src/plugins/my-plugin/admin/src/index.js
+
+import Link from './components/Link'
+
+export default {
+  bootstrap(app){
+    app.injectContentManagerComponent('editView', 'right-links', {
+      name: 'content-type-builder.link',
+      Component: Link,
+    });
+  }
+}
+```
+
+:::
+
+## Using the `Initializer` component
+
+::: strapi v3/v4 comparison
+In both Strapi v3 and v4, the `Initializer` component is generated by default when a plugin is created. `Initializer` could be used to execute some logic when the application is loading, for instance to dispatch an action after the user logs in into the admin panel.
+
+In Strapi v4, the `Initializer` component code has changed and uses the `useRef` and `useEffect`  React hooks to dispatch actions.
+:::
+
+To migrate to Strapi v4, make sure the plugin uses the latest `Initializer` code, which can be copied and pasted from the following code example:
+
+```jsx
+// path: ./src/plugins/my-plugin/admin/src/components/Initializer/index.js
+
+import { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import pluginId from '../../pluginId';
+
+const Initializer = ({ setPlugin }) => {
```

Comment:
Do you think that we should a comment explaining when it is useful to use such component?

For instance:

```
The Initializer component is useful to use when you need to store a global state in your application that will need to be loaded before the app is rendered using redux.
```